src/Symfony/Component/Mailer/Mailer.php

```diff
@@ -36,6 +38,10 @@ public function __construct(TransportInterface $transport, MessageBusInterface $
 
     public function send(RawMessage $message, Envelope $envelope = null): void
     {
+        if ($message instanceof Message && '1' === $message->getHeaders()->getHeaderBody('X-Unsent')) {
+            throw new LogicException('Cannot send messages marked as "draft".');
+        }
```

Comment:
Is there a better place to put this?

Reply:
I wonder if it were better to type-check for `DraftEmail` instead. This block would trigger an exception on messages that are currently accepted by the mailer.

Reply:
I would have expected the exception in `ensureValidity` instead. I suppose there is a reason that makes it impossible?
Here, it means that using the transports directly won't share this logic.

Reply:
> I suppose there is a reason that makes it impossible?

Yes, `ensureValidity` must not throw an exception when converting message to string.

> Here, it means that using the transports directly won't share this logic.

Yeah, or if someone has their own `MailerInterface` implementation.

Reply:
What about throwing in `SentMessage::__construct()`?

Reply:
I believe ad833aaaedf474b570d6c43b5ded28adbcf41cce resolves this.